classes/patreon_login.php

```diff
@@ -14,72 +14,208 @@ public static function updateExistingUser($user_id, $user_response, $tokens) {
 		update_user_meta($user_id, 'patreon_refresh_token', $tokens['refresh_token']);
 		update_user_meta($user_id, 'patreon_access_token', $tokens['access_token']);
 		update_user_meta($user_id, 'patreon_user', $user_response['data']['attributes']['vanity']);
+		update_user_meta($user_id, 'patreon_user_id', $user_response['data']['id']);
+		update_user_meta($user_id, 'patreon_last_logged_in', time());
 		update_user_meta($user_id, 'patreon_created', $user_response['data']['attributes']['created']);
 		update_user_meta($user_id, 'user_firstname', $user_response['data']['attributes']['first_name']);
 		update_user_meta($user_id, 'user_lastname', $user_response['data']['attributes']['last_name']);
 		update_user_meta($user_id, 'patreon_token_minted', microtime());
+		update_user_meta($user_id, 'patreon_token_expires_in', $tokens['expires_in']);
 
 	}
 
-	public static function updateLoggedInUser($user_response, $tokens, $redirect = false) {
+	public static function updateLoggedInUserForStrictoAuth($user_response, $tokens, $redirect = false) {
 
 		$user = wp_get_current_user();
 
 		if(0 == $user->ID) {
+			
+			$redirect = add_query_arg( 'patreon_message', 'patreon_cant_login_strict_oauth', $redirect);
+			wp_redirect($redirect);
+			exit;			
 		} else {
 			/* update user meta data with patreon data */
 			self::updateExistingUser($user->ID, $user_response, $tokens);
+			wp_redirect($redirect);
+			exit;				
 		}
 
 		if($redirect == false || is_null($redirect) ) {
 			wp_redirect(home_url());
 			exit;
 		} else {
-			wp_redirect( get_permalink($redirect) );
+			wp_redirect( $redirect );
 			exit;
 		}
 
 	}
 
-	public static function createUserFromPatreon($user_response, $tokens, $redirect = false) {
+	public static function checkTokenExpiration($user_id=false) {
+
+		if($user_id)
+		{
+			$user = get_user_by('ID',$user_id);
+		}
+		else
+		{
```

Comment:
Unify the style of `if` `else` statements 
instead of
```
if (statement) 
{
    // Do a thing
} 
else 
{ 
    // Do another thing
}
```
Do this
```
if (statement) {
    // Do a thing
} else { 
    // Do another thing
}
```

Reply:
Styling and typo commit:

https://github.com/codebard/patreon-wordpress/commit/2c5f9c2ff00709f2acd7e6cc5f25915e61b8f02a